codegen-winter/src/air/public_inputs.rs

```diff
@@ -0,0 +1,28 @@
+use ir::AirIR;
+
+use super::Scope;
+
+/// Updates the provided scope with a public inputs.
+pub(super) fn add_public_inputs_struct(scope: &mut Scope, ir: &AirIR) {
+    let name = "PublicInputs";
+    // define the custom Air struct.
+    let pub_inputs_struct = scope.new_struct(name).vis("pub");
+
+    for (pub_input, pub_input_size) in ir.public_inputs() {
+        pub_inputs_struct.field(pub_input, format!("[Felt; {}]", pub_input_size));
+    }
+
+    // add the custom Air implementation block
+    let base_impl = scope.new_impl(name);
+
+    // add a constructor for public inputs
+    base_impl
+        .new_fn("new")
+        .arg("program_hash", "Digest")
+        .arg("stack_inputs", "Vec<Felt>")
+        .arg("stack_outputs", "Vec<u64>")
+        .arg("overflow_addrs", "Vec<u64>")
+        .vis("pub")
+        .ret("Self")
+        .line("Self { program_hash, stack_inputs, stack_outputs, overflow_addrs }");
```

Comment:
I think it's just an oversight, but these argument names shouldn't be hard-coded, and all of the types should be the same: `Vec<Felt>`

Reply:
Thanks, fixed. Should the types be fixed size arrays or vectors?

Reply:
Actually you can make them fixed size arrays. Good point :)